runtime/vm.go

```diff
@@ -642,9 +642,34 @@ func (v *VM) run() {
 
 			switch callee := value.(type) {
 			case *objects.Closure:
+				if callee.Fn.VarArgs {
+					// if the closure is variadic,
+					// roll up all variadic parameters into an array
+					realArgs := callee.Fn.NumParameters - 1
+					varArgs := numArgs - realArgs
+					if varArgs < 0 {
+						goto wrongNumberOfParametersClosure
+					}
+					numArgs = realArgs + 1 // must come after varArgs check
+					args := make([]objects.Object, 0, varArgs)
+					spStart := v.sp - varArgs
+					for i := spStart; i < v.sp; i++ {
+						args = append(args, v.stack[i])
+					}
+
+					v.stack[spStart] = &objects.Array{Value: args}
+					v.sp = spStart + 1
+				}
+
+			wrongNumberOfParametersClosure:
```

Comment:
we can remove `goto` and don't need `append` for `varArgs`:

```golang
if callee.Fn.VarArgs {
	// if the closure is variadic,
	// roll up all variadic parameters into an array
	realArgs := callee.Fn.NumParameters - 1
	varArgs := numArgs - realArgs
	if varArgs >= 0 {
		numArgs = realArgs + 1
		args := make([]objects.Object, varArgs)
		spStart := v.sp - varArgs
		for i := spStart; i < v.sp; i++ {
			args[i-spStart] = v.stack[i]
		}
		v.stack[spStart] = &objects.Array{Value: args}
		v.sp = spStart + 1
	}
}
```

Reply:
not sure if this affects benchmark results or not.

Reply:
agreed, fixed